mh/constants.py

```diff
@@ -180,21 +185,16 @@ def make_binary_npc_greeters(is_jap=False):
         material_shop = b"Prices are normal."
         event_quests = b""
 
-    QUEST_LIST = (
-        QUEST_EVENT_JUMP_FOUR_JAGGI,
-        QUEST_EVENT_BLOOD_SPORT,
-        QUEST_EVENT_MERCY_MISSION,
-        QUEST_EVENT_THE_PHANTOM_URAGAAN,
-        QUEST_EVENT_WORLD_EATER,
-        QUEST_EVENT_WHERE_GODS_FEAR_TO_TREAD
-    )
+    QUEST_LIST = []
+    ARENA_LIST = []
 
-    ARENA_LIST = (
-        GRUDGE_MATCH_ROYAL_LUDROTH,
-        GRUDGE_MATCH_BIRD_BRUTE,
-        GRUDGE_MATCH_TWO_FLAMES,
-        GRUDGE_MATCH_LAND_LORDS
-    )
+    for quest in binary_loader.get_quest_dicts(temporal_slot):
+        if quest is None:
+            continue
+        if is_arena_quest(generate_flags(*(quest['quest_info']['flags']))):
+            ARENA_LIST.append(quest)
+        else:
+            QUEST_LIST.append(quest)
```

Comment:
The quest loader should returns the `QUEST_LIST` and `ARENA_LIST` instead so the logic is kept there. Same remark regarding the binary loader so we don't need to pass it as parameter.

Reply:
This logic has been moved there